### ISSUE TITLE
Type STRING in Database instead of INTEGER/TEXT

### ISSUE THREAD

**Issue by tobiasKaminsky** [nextcloud/android#1821]
AndroidStudio suggests not to use STRING as this type can truncate leading zeros, instead Integer or Text should be used.
As this is done on upgrade from very old databases or on new installation I guess we can change this without side-effects.
Is FILESYSTEM_SYNCED_FOLDER_ID integer only? Or should we use text?

https://github.com/nextcloud/android/blob/f9983558c97bd0ef61158b9dc93ab89d20e60e2a/src/main/java/com/owncloud/android/providers/FileContentProvider.java#L1363

@mario as you wrote the code and probably know best

**Comment by AndyScherzinger** [comment]
Are you sure this can "just be done"? installations that are already past that DB number would then have a different data type set... So Best would actually be to create a new DB version which upgrades the column.

**Comment by tobiasKaminsky** [comment]
Good point, so simply changing it is not the best way. So we do an upgrade and need to convert existing entries.
But we still need to know if it should change to TEXT or INTEGER.

**Comment by AndyScherzinger** [comment]
True, so pining @mario for input/feedback :)

**Comment by mario** [comment]
INTEGER is fine. :)

**Comment by joshtrichards** [comment]
Things have moved around a bit, but still the case today:

https://github.com/nextcloud/android/blob/2696bdca522ec9968517a1a9f240f5ebdeb85c55/app/src/main/java/com/nextcloud/client/database/entity/FilesystemEntity.kt#L28-L29
